tests/queryScope.js

```diff
@@ -0,0 +1,26 @@
+
+define([
+    "esquery",
+    "jstestr/assert",
+    "jstestr/test",
+    "./fixtures/nestedFunctions",
+    "./fixtures/nestedFunctionsWithReturns",
+], function (esquery, assert, test, nestedFunctions, nestedFunctionsWithReturns) {
+
+    test.defineSuite("scope selector", {
+
+        "select only first level function": function () {
```

Comment:
Is this name a valid description? I'm really confused about the difference between `:scope` and `:root` 😅

Reply:
I need to check that, but the difference between `:scope` and `:root` is pretty simple.

If we look at the css documentation ([root](https://developer.mozilla.org/en-US/docs/Web/CSS/:root)) for selector `:root` it says that in a stylesheet `:root` select the root element of the tree.

So, in our case we should be able to do this : 
```
const example = `
function hi(){}
`
// We are selecting the function
const identifier = esquery(example, 'Identifier');

// From the identifier hi we are selecting the `:root` element
const root = esquery(identifier, ':root');

// From the root we select the identifier` hi` again and log it's name
console.log(esquery(root, 'Identifier').name)
// => hi
```

For example with `:scope` we can do this : 
```
const example = `
function hi(){}
`
// We are selecting the identifier `hi`
const functionDeclaration = esquery(example, 'FunctionDeclaration')[0];

// From the identifier hi we are selecting the direct child of `:scope` which are `BlockStatement`.
const block = esquery(identifier, ':scope > BlockStatement');

// One block statement has been selected
console.log(block.length)
// => 1
```

If we try to do the same thing with `:root`, it can't match any `BlockStatement` : 
```
const example = `
function hi(){}
`
// We are selecting the function
const functionDeclaration = esquery(example, 'FunctionDeclaration')[0];

// From the function we are selecting the direct child of `:root` which are `BlockStatement`.
const block = esquery(identifier, ':root > BlockStatement');

// There is no `BlockStatement` has direct child of the root document
console.log(block.length)
// => 0
```

For me it should work like this, I will recheck the tests to see if it's match the exact explanation below.

Reply:
After checking the test again and reading the code, it does not match what I was thinking. After a little bit of investigation, the `:scope` and `:root` selector seems to be equivalent.
I can't see why there is a difference in the code between `:scope` and `:root` thought...